### PR TITLE
Make code pylint compatible and integrate pylint into tests

### PR DESCRIPTION
**Pull request by cosama**
Fixes issue #125.

I started working on the core modules, which should be good now. Have a look if this is pleasing and is a style that we can all agree to.

Before it can be pulled in we have to:
- [x] Fix core module
- [ ] Fix parser module
- [ ] Fix tools module
- [ ] Deal with fixme's and TODO's
- [ ] Figure out how to check tests
- [ ] Pull in most recent develop
- [ ] Add pylint support to tests

### CODE REVIEW

File: becquerel/core/plotting.py [hunk 1]
```diff
@@ -532,3 +466,78 @@ def linthreshy(self):
         min_ind = np.argmin(np.abs(self._ydata[self._ydata != 0]))
         delta_y = np.abs(self._ydata - self._ydata[min_ind])
         return np.min(delta_y[delta_y > 0])
+
+
+def get_channel_edges(channels):
```
Comment (jccurtis) [inline becquerel/core/plotting.py]: @cosama why did you remove the `staticmethods` in favor of funcs? Is there a style guideline regarding this modification?

Comment (cosama) [reply becquerel/core/plotting.py]: Yeah it is in the google style guide to rather use module methods. Not sure why though. Do you have any ideas?

Comment (jccurtis) [reply becquerel/core/plotting.py]: Nope. Let’s follow the google style guide on this. Thanks!

**Comment by jccurtis** [comment]
We are waiting to fixing linting until after v0.2.0

**Comment by jccurtis** [comment]
This branch will need to be remade (maybe a new PR?) Moving to `v1.0.0`

**Comment by markbandstra** [comment]
Made obsolete by #221 and #220
